kizhin.evgeniy/P6

sdjklfjsdlkfjsdklfj opened the pull request:
13

Volgarenok commented:
@sdjklfjsdlkfjsdklfj Всё здорово, только с глобальными переменными это точно никуда не годится

Задание на рекурсию и это просто упражнение на рекурсию. Если бы была задача на парсинг, тогда мы бы делали конечный автомат, поток токенов и так далее

Если хотите задачу под эту тему, я вам её дам, но давайте тут просто сделаем, что от нас хотят

Commit:
Initial commit for P6: create main

Implemented main function which firslty reads string from standart
input via function readString from common library and then calls
isExpression function that will check is given string a valid
expression according to variant 13 grammatic

Commit:
Implement is_expression module

Volgarenok commented:
@sdjklfjsdlkfjsdklfj Почему у вас force-pushed всё время?

Volgarenok commented on kizhin.evgeniy/P6/is_expression.cpp:
```diff
@@ -0,0 +1,75 @@
+const char* kizhin::isIdentifier(const char* position)
+{
+  const char* valid = "abcxyz";
+  for (const char* i = valid; *i != '\0'; ++i) {
+    if (*position == *i) {
+      return ++position;
+    }
```
Циклов в лабораторной на рекурсию быть не должно

sdjklfjsdlkfjsdklfj commented:
> @sdjklfjsdlkfjsdklfj Почему у вас force-pushed всё время?

Сложно ответить за force-pushed в других PR, но конкретно в этом их было два и вот их объяснения:

1. Первый связан с ошибкой в названии переменной, обнаруженной уже после открытия PR. Исправление внесено в исходный коммит для сохранения аккуратной истории.
2. Второй связан с тем, что после ревью текущий код оказался непригодным для доработки. Было принято решение удалить прежние коммиты и начать работу заново, чтобы история PR оставалась чистой.

Приношу извинения за неудобства. Постараюсь избегать правок после открытия PR до первого ревью, а также тщательнее проверять коммиты перед отправкой.

Commit:
Remove for-loop from IsIndetifier function

Used recursion instead according task requrements

Commit:
Merge branch 'master' of https://github.com/Volgarenok/spbspu-labs-2024-aap-a into kizhin.evgeniy/P6

Volgarenok commented:
> * Первый связан с ошибкой в названии переменной, обнаруженной уже после открытия PR. Исправление внесено в исходный коммит для сохранения аккуратной истории.
> * Второй связан с тем, что после ревью текущий код оказался непригодным для доработки. Было принято решение удалить прежние коммиты и начать работу заново, чтобы история PR оставалась чистой.

Ну так всё-таки лучше не делать. Аккуратная история обычно обеспечивается свёрткой комитов при мёрже. Само собой история разработки "фичи" стирается, но логически история проекта выглядит структурированней
